### PR TITLE
Add train support, upgrade versions

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -53,6 +118,7 @@ configure(subprojects.findAll { it.name.startsWith("maven-") }) {
     def mvnw = System.getProperty('os.name').toLowerCase(Locale.US).contains('windows') ?
             [ 'cmd', '/c', 'mvnw.cmd' ] : [ './mvnw' ]
 
+    mvnw += "-Dkotlin.version=$kotlin_version"
```
Comment (cy6erGn0m) [inline build.gradle]: Shouldn't we override all library versions as well? Or we only use ktor.version in maven examples?

File: build.gradle [hunk 2]
```diff
@@ -79,19 +147,27 @@ configure(subprojects.findAll { it.name.startsWith("maven-") }) {
...
-            maven { url 'https://dl.bintray.com/kotlin/kotlin-eap' }
-        }
-        dependencies {
-            classpath "com.android.tools.build:gradle:3.4.1"
+
+if (build_snapshot_train) {
+    println "Hacking test tasks, removing stress and flaky tests"
+    allprojects {
+        tasks.withType(Test).all {
+            // Add: exclude '**/*TestName*' here
```
Comment (cy6erGn0m) [inline build.gradle]: So we do nothing? Or force configuring?

File: build.gradle [hunk 3]
```diff
@@ -79,19 +147,27 @@ configure(subprojects.findAll { it.name.startsWith("maven-") }) {
...
+    println "Manifest of kotlin-compiler-embeddable.jar for coroutines"
+    configure(subprojects.findAll { it.name == "kotlinx-coroutines-core" }) {
+        configurations.matching { it.name == "kotlinCompilerClasspath" }.all {
+            resolvedConfiguration.getFiles().findAll { it.name.contains("kotlin-compiler-embeddable") }.each {
+                def manifest = zipTree(it).matching {
+                    include 'META-INF/MANIFEST.MF'
+                }.getFiles().first()
+
+                manifest.readLines().each {
+                    println it
```
Comment (cy6erGn0m) [inline build.gradle]: debug?

File: other/maven-google-appengine-standard/pom.xml [hunk 4]
```diff
@@ -11,8 +11,8 @@
     <artifactId>maven-google-appengine-standard</artifactId>
 
     <properties>
-        <kotlin.version>1.3.61</kotlin.version>
-        <ktor.version>1.3.0</ktor.version>
+        <kotlin.version></kotlin.version>
```
Comment (cy6erGn0m) [inline other/maven-google-appengine-standard/pom.xml]: This is not good since users copy-paste this and may be confused

File: build.gradle (hunk 1 above)
Comment (e5l) [reply build.gradle]: We use only kotlin and ktor dependencies in maven

File: build.gradle (hunk 2 above)
Comment (e5l) [reply build.gradle]: We keep the configuration as-is for now, so it's easy to add flaky tests here in future

File: build.gradle (hunk 3 above)
Comment (e5l) [reply build.gradle]: Yep, for train purposes. We have the same line in ktor.

File: other/maven-google-appengine-standard/pom.xml (hunk 4 above)
Comment (e5l) [reply other/maven-google-appengine-standard/pom.xml]: It's only for one purpose: fail if no CLI argument. There was a problem here: no CLI arguments were provided, and the samples always use old versions. 

Maybe we should add XML comment here

**Pull request merged**